src/main/kotlin/it/pagopa/wallet/documents/wallets/Wallet.kt

```diff
@@ -1,38 +1,42 @@
 package it.pagopa.wallet.documents.wallets
 
+import com.azure.spring.data.cosmos.core.mapping.PartitionKey
 import it.pagopa.generated.wallet.model.WalletStatusDto
 import it.pagopa.wallet.documents.wallets.details.WalletDetails
 import it.pagopa.wallet.domain.wallets.*
-import it.pagopa.wallet.domain.wallets.Wallet
+import it.pagopa.wallet.domain.wallets.Wallet as WalletDomain
 import java.time.Instant
 import java.util.*
+import org.springframework.data.annotation.CreatedDate
+import org.springframework.data.annotation.Id
+import org.springframework.data.annotation.LastModifiedDate
+import org.springframework.data.annotation.Version
 import org.springframework.data.mongodb.core.mapping.Document
 
 @Document("wallets")
 data class Wallet(
-    val id: String,
+    @PartitionKey val walletId: String,
     val userId: String,
     val status: String,
-    val creationDate: String,
-    val updateDate: String,
     val paymentMethodId: String,
     val paymentInstrumentId: String?,
     val contractId: String?,
     val applications: List<Application>,
-    val details: WalletDetails<*>?
+    val details: WalletDetails<*>?,
+    @CreatedDate val creationDate: Instant? = null,
```

Comment:
`creationDate`, `updateDate` and `version` are not optional, there is no valid document/wallet without these three fields, so they must be mandatory and initialized only when creating a new wallet